src/Flash.js

```diff
@@ -1,44 +1,87 @@
+import React from 'react'
 import PropTypes from 'prop-types'
 import styled from 'styled-components'
+import {variant} from 'styled-system'
 import {COMMON, get} from './constants'
 import theme from './theme'
 import sx from './sx'
+import deprecate from './utils/deprecate'
 
 const schemeMap = {
-  green: {color: 'colors.green.8', bg: 'colors.green.1'},
-  red: {color: 'colors.red.9', bg: 'colors.red.1'},
-  yellow: {color: 'colors.yellow.9', bg: 'colors.yellow.1'},
-  base: {color: 'colors.blue.8', bg: 'colors.blue.1'}
+  red: 'danger',
+  blue: 'default',
+  yellow: 'warning',
+  green: 'success'
 }
 
-const Flash = styled.div`
+const variants = variant({
+  variants: {
+    success: {
+      borderColor: 'flash.success.border',
+      bg: 'flash.success.bg'
+    },
+    danger: {
+      borderColor: 'flash.danger.border',
+      bg: 'flash.danger.bg'
+    },
+    warning: {
+      borderColor: 'flash.warning.border',
+      bg: 'flash.warning.bg'
+    },
+    default: {
+      borderColor: 'flash.default.border',
+      bg: 'flash.default.bg'
+    }
```

Comment:
It looks like we've basically defined the variant structure in the theme. So I think we can replace this object with `scale: 'flash'`:

```suggestion
    scale: 'flash'
```

We'd also have to change `flash.default.border` to `flash.default.borderColor`.

Here's are the styled-system docs about the `scale` option: https://styled-system.com/variants/#themeable-variants

Reply:
This doesn't seem to work 🤔